src/kernels/execution/cellExecutionMessageHandler.ts

```diff
@@ -627,6 +692,9 @@ export class CellExecutionMessageHandler implements IDisposable {
         if (outputsAfterWidget.length === 0 && outputToAppend) {
             // No need to replace everything, just append what we need.
             task?.appendOutput(outputToAppend, cell).then(noop, noop);
+        } else if (clearWidgetOutput && !outputToAppend && outputsAfterWidget.length === 0) {
+            // We're supposed to clear the output after a specific widget, but there are no other outputs after the widget.
```

Comment:
Minor optimization, previously we were re-rendering the same output even when there were no changes.